src/utils/utils.ts

```diff
@@ -0,0 +1,14 @@
+import dayjs, { Dayjs } from 'dayjs';
+
+export const isServer = typeof window === 'undefined';
+export const calculateTimeRemaining = (createdAt: Dayjs) => {
+  const currentTime = dayjs();
+  // add not exact 15 minutes but 14:50 to consider network delays
+  const diffInSeconds = dayjs(createdAt) // I don't know but createAt is passed as string
+    .add(14, 'minutes')
+    .add(50, 'seconds')
+    .diff(currentTime, 'second');
```

Comment:
```suggestion
  const editingLimit = dayjs(createdAt).add(14, 'minutes').add(50, 'seconds');
  const diffInSeconds = editingLimit.diff(currentTime, 'second');
```

server side에서는 Dayjs 객체를 프롭으로 넘길 수 없기 때문에 string으로 전달이 됩니다
dayjs.Dayjs | string으로 타입을 다 받아서 dayjs로 한번 래핑해주는게 좋습니당

Reply:
와우 그래서였군요